example/hook.go

```diff
@@ -0,0 +1,55 @@
+package example
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Makpoc/gomigen/types"
+)
+
+type ctxKey string
+
+const (
+	startTimeKey ctxKey = "startTime"
+)
+
+// LogHook is an example of a hook that logs messages on entry and exit and the time interval
+// between they are called.
+type LogHook struct {
+	logger *log.Logger
+}
+
+var _ types.Hook = (*LogHook)(nil)
+
+func New(log *log.Logger) *LogHook {
+	return &LogHook{
+		logger: log,
+	}
+}
+
+func (l *LogHook) OnEntry(ctx context.Context, info types.MethodInfo) context.Context {
+	startTime := time.Now()
+	ctx = context.WithValue(ctx, startTimeKey, startTime)
+	l.logger.Printf("Entered %s at %s",
+		formatMethodInfo(info), startTime,
+	)
+	return ctx
+}
+
+func (l *LogHook) OnExit(ctx context.Context, info types.MethodInfo, err error) {
+	duration := "unknown"
+	startTime := ctx.Value(startTimeKey)
+	if startTime != nil {
```

Comment:
If the context value has been set in the OnEnter hook, is there any theoretical chance that it will not be set here?
My understanding is that even if the service method does not accept a Context, the framework generates a temp one and ensures it is passed between the two functions.
By keeping this `if`, you are creating questions in the readers mind. Also, if you remove it, you can save a few lines of code (e.g. the default duration initialization)

Reply:
Changed. Also using `err` in the example now (if set).